maxdiff/print_unicode.py

```diff
@@ -0,0 +1,14 @@
+"""
+Module that shares a function to gracefully handle printing unicode characters 
+if the user has the PYTHONIOENCODING environment vairable set to 'ascii'. 
+In that case, replace any characters that can't be encoded with ?.
+"""
+
+
+def print_unicode_string(s):
+    try:
+        encoded_string = s.encode("utf-8", "replace")
+        print(s)
+    except UnicodeEncodeError:
+        encoded_string = s.encode("ascii", "replace")
```

Comment:
What happens here if encoding fails? Are we also happy with ? for any character that it needs to replace?

Essentially, does this cover all the bases of an encoding being not UTF-8?

Reply:
I agree, if the change in a patch would only be to use a different unicode character, we wouldn't see a diff with this approach. 

I'll replace the error handling flag to `backslashreplace`, which should convert UTF-8 characters to ascii code; e.g. the one used in the test patch converts to `\U0001f60b`.